vendor/github.com/emicklei/go-restful/v3/jsr311.go

```diff
@@ -155,7 +155,7 @@ func (r RouterJSR311) detectRoute(routes []Route, httpRequest *http.Request) (*R
 		method, length := httpRequest.Method, httpRequest.Header.Get("Content-Length")
 		if (method == http.MethodPost ||
 			method == http.MethodPut ||
-			method == http.MethodPatch) && length == "" {
+			method == http.MethodPatch) && (length == "" || length == "0") {
```

Comment:
uh... this looks incorrect at first glance... is an explicitly empty (zero-length) body invalid?

things like https://lists.w3.org/Archives/Public/ietf-http-wg/2010JulSep/0276.html led me to think a zero-length body on put/post/patch was valid

Reply:
commented at https://github.com/emicklei/go-restful/pull/543#issuecomment-2495000742